src/gwt/src/org/rstudio/core/client/command/ShortcutManager.java

```diff
@@ -337,6 +359,25 @@ public int compare(ShortcutInfo o1, ShortcutInfo o2)
    
    private boolean handleKeyDown(NativeEvent event)
    {
+      // Don't handle the (synthetic) copy, cut, paste keys
```

Comment:
Just to make it clear what's going on here, the goal is to allow e.g. Copy to succeed when executed from the menu. This is done in the following way:

- The 'copyDummy' command emits an `EditEvent(true)`, which Ace listens for and uses to suppress custom `C-c` handlers;

- A `C-c` key event is emitted from the server side (note that this is done asynchronously),

- The `ShortcutManager` receives that key event, but lets it pass through (so that it can be handled as just a regular copy). It fires a (deferred) `EditEvent(false)` event, to signal that the copy has finished and custom Ace bindings can be restored.